### PR TITLE
Feature/observability custom panels 

### PR DESCRIPTION
**Pull request by ps48**
### Description
Added initial placeholder for custom operational panels 

### Issues resolved
Custom Panels - Home Page Frontend #68

### Additional notes:
1. Backend used is dummy as of now. 
2. Tests to be added later 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/routes/custom_panels_router.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  
+  // NOTE: Currently the API calls are dummy and are not connected to esclient.
+  // Fetch all the custom panels available
+  router.get(
+    {
+      path: `${API_PREFIX}/`,
+      validate: {},
+    },
+    async (context, request, response): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
+      let panelsList = [{
```
Comment (mengweieric) [inline server/routes/custom_panels_router.ts:31]: Seems like you are not changing the value of panelsList, could you use const?

File: public/components/custom_panels/custom_panel_view.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { EuiBreadcrumb } from '@elastic/eui';
```
Comment (joshuali925) [inline public/components/custom_panels/custom_panel_view.tsx:1]: license missing

File: public/components/custom_panels/helpers/custom_modals/custom_input_modal.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (joshuali925) [inline public/components/custom_panels/helpers/custom_modals/custom_input_modal.tsx:13]: FYI i don't think this license header is required unless the same file existed in ODFE repo

File: public/components/custom_panels/custom_panel_view.tsx (hunk 2 above)
Comment (mengweieric) [inline public/components/custom_panels/custom_panel_view.tsx:1]: Could you add the license?

File: public/components/custom_panels/home.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      .get(`${CUSTOM_PANELS_API_PREFIX}/`)
+      .then((res) => setcustomPanelData(res.panels))
+      .catch((err) => {
+        console.error('Issue in fetching the custom operational panels', err.body.message);
+      });
+  };
+
+  // Creates a new CustomPanel
+  const createCustomPanel = (newCustomPanelName: string) => {
+    if (newCustomPanelName.length >= 50 || newCustomPanelName.length === 0) {
```
Comment (joshuali925) [inline public/components/custom_panels/home.tsx:63]: maybe move the validation logic into some utils module? so it's not defined multiple places and easier to change

**Review by joshuali925 — APPROVED** [review]
minor comments

File: public/components/custom_panels/custom_panel_view.tsx (hunk 2 above)
Comment (ps48) [reply public/components/custom_panels/custom_panel_view.tsx:1]: will add it 👍🏽

File: public/components/custom_panels/helpers/custom_modals/custom_input_modal.tsx (hunk 3 above)
Comment (ps48) [reply public/components/custom_panels/helpers/custom_modals/custom_input_modal.tsx:13]: Will remove this one. The file has many things common with ODFE repo "notebooks" modal container. But will add more modal components going forward.

File: public/components/custom_panels/home.tsx (hunk 4 above)
Comment (ps48) [reply public/components/custom_panels/home.tsx:63]: Oh yes this makes sense.  Will move this function to helpers.

File: server/routes/custom_panels_router.ts (hunk 1 above)
Comment (ps48) [reply server/routes/custom_panels_router.ts:31]: For sure, this should be const.

File: public/components/custom_panels/custom_panel_view.tsx (hunk 2 above)
Comment (ps48) [reply public/components/custom_panels/custom_panel_view.tsx:1]: Thanks for the catch 👍🏽

**Review by mengweieric — APPROVED** [review]
LGTM, left minor comments

**Comment by ps48** [comment]
Edited files based on review comments.

**Pull request merged**